posydon/unit_tests/utils/test_configfile.py

```diff
@@ -0,0 +1,290 @@
+        with monkeypatch.context() as mp:
+            # replace parse to check iteration
+            mp.setattr(totest.ast, "parse", mock_parse)
+            # replace content of test.ini to check Error
+            with open(test_path, "w") as test_file:
+                test_file.write("[Unit Test]\n")
+                test_file.write("test_value = 'Test'\n")
+            with raises(KeyError, match="'run_parameters'"):
+                totest.parse_inifile(test_path)
+
+
+class TestConfigFile:
+    @fixture
+    def ConfigFile(self):
+        # initialize an instance of the class with defaults
+        ConfigFile = totest.ConfigFile()
+        return ConfigFile
+
+    @fixture
+    def test_path(self, tmp_path):
+        # a temporary path for testing
+        return totest.os.path.join(tmp_path, "test.json")
+
+    @fixture
+    def test_json(self, test_path):
+        # a temporary json file for testing
+        with open(test_path, "w") as test_file:
+            test_file.write('{\n    "Unit": "Test"\n}\n')
+        return
+
+    @fixture
+    def test_entries(self):
+        # entries for testing
+        return {'Unit1': "Test", 'Unit2': "Again"}
+
+    @fixture
+    def ConfigFile_filled(self, test_entries):
+        # initialize an instance of the class with defaults
+        ConfigFile = totest.ConfigFile()
+        ConfigFile.update(test_entries)
+        return ConfigFile
+
+    # test the ConfigFile class
+    def test_init(self, ConfigFile, tmp_path):
+        assert isroutine(ConfigFile.__init__)
+        # check that the instance is of correct type and all code in the
+        # __init__ got executed: the elements are created and initialized
+        assert isinstance(ConfigFile, totest.ConfigFile)
+        assert ConfigFile.entries == {}
+        assert ConfigFile.path is None
+        # error on directory
+        with raises(IsADirectoryError, match="Is a directory: '"+\
+                                             str(tmp_path)+"'"):
+            totest.ConfigFile(path=tmp_path)
+        # non-existing path
+        test_path = totest.os.path.join(tmp_path, "does_not_exist.test")
+        test_ConfigFile = totest.ConfigFile(test_path)
+        assert test_ConfigFile.path == test_path
+
+    def test_deepcopy(self, ConfigFile):
+        assert isroutine(ConfigFile.deepcopy)
+        ConfigFile.entries['Unit'] = "Test"
+        ConfigFile.path = "Test_path"
+        test_ConfigFile = ConfigFile.deepcopy()
+        # not same object
+        assert test_ConfigFile is not ConfigFile
+        # but same content
+        assert test_ConfigFile.entries == ConfigFile.entries
+        assert test_ConfigFile.path == ConfigFile.path
+
+    def test_serialize(self, ConfigFile):
+        assert isroutine(ConfigFile._serialize)
+        # serialize numpy array
+        np_array = totest.np.array([2, 1, 0])
+        assert ConfigFile._serialize(data=np_array) == [2, 1, 0]
+        # other data don't give a return
```

Comment:
This function is requested by json.dump to deal with objects json doesn't know about, here we define a rule how to deal with ndarrays.
It works for any ndarray as long as it can be converted to a list.